rust/lance-encoding/src/encoder.rs

```diff
@@ -430,6 +475,83 @@ fn check_dict_encoding(arrays: &[ArrayRef], threshold: u64) -> bool {
     true
 }
 
+fn check_fixed_size_encoding(arrays: &[ArrayRef], byte_width: &mut u64) -> bool {
```

Comment:
Output parameters should generally be avoided if possible.  They are sometimes inevitable (e.g. in a visitor pattern / recursive calling pattern) but I don't think this one is.  You could return `Option<u64>`.